ml_tools/eolearn/ml_tools/TDigestTask.py

```diff
@@ -0,0 +1,188 @@
+        )
+
+        if len(self.in_feature) != len(self.out_feature):
+            raise ValueError(
+                f"The number of input ({len(self.in_feature)}) and output features ({len(self.out_feature)}) must"
+                " match."
+            )
+
+    def execute(self, eopatch: EOPatch) -> EOPatch:
+        """
+        Execute method that computes the TDigest of the chosen features.
+
+        :param eopatch: EOPatch which the chosen input feature already exists
+        """
+
+        if self.mode == "standard":
+            _process_standard(
+                in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch, pixelwise=self.pixelwise
+            )
+
+        elif self.mode == "timewise":
+            _process_timewise(
+                in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch, pixelwise=self.pixelwise
+            )
+
+        elif self.mode == "monthly":
+            _process_monthly(
+                in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch, pixelwise=self.pixelwise
+            )
+
+        elif self.mode == "total":
+            _process_total(in_feature=self.in_feature, out_feature=self.out_feature, eopatch=eopatch)
+
+        return eopatch
+
+
+# auxiliary
+ModeTypes = Literal["standard", "timewise", "monthly", "total"]
+
+
+def _is_input_ftype(feature_type: FeatureType, mode: ModeTypes) -> bool:
+    if mode == "standard":
+        return feature_type.is_image()
+    if mode in ("timewise", "monthly"):
+        return feature_type in [FeatureType.DATA, FeatureType.MASK]
+    return True
+
+
+def _is_output_ftype(feature_type: FeatureType, mode: ModeTypes, pixelwise: bool) -> bool:
+    if mode == "standard":
+        return feature_type == (FeatureType.DATA_TIMELESS if pixelwise else FeatureType.SCALAR_TIMELESS)
+
+    if mode in ("timewise", "monthly"):
+        return feature_type == (FeatureType.DATA if pixelwise else FeatureType.SCALAR)
+
+    return feature_type == FeatureType.SCALAR_TIMELESS
+
+
+def _looper(in_feature: FeaturesSpecification, out_feature: FeaturesSpecification, eopatch: EOPatch) -> tuple:
+    for in_feature_, out_feature_ in zip(in_feature, out_feature):
+        shape = np.array(eopatch[in_feature_].shape)
+        yield in_feature_, out_feature_, shape
+
+
+def _process_standard(
+    in_feature: FeaturesSpecification, out_feature: FeaturesSpecification, eopatch: EOPatch, pixelwise: bool
+) -> None:
+    for in_feature_, out_feature_, shape in _looper(in_feature=in_feature, out_feature=out_feature, eopatch=eopatch):
+        if pixelwise:
+            eopatch[out_feature_] = np.empty(shape[-3:], dtype=object)
+            for i, j, k in product(range(shape[-3]), range(shape[-2]), range(shape[-1])):
+                eopatch[out_feature_][i, j, k] = td.TDigest()
+                eopatch[out_feature_][i, j, k].batch_update(eopatch[in_feature_][..., i, j, k].flatten())
```

Comment:
this comes up often. Perhaps consider something along the lines of
```python
def get_tdigest(values: np.ndarray) -> td.TDigest:
    result = td.TDigest()
    result.batch_update(values.flatten())
    return result
```
This way you have just:
```python
out_array[i, j, k] = get_tdigest(in_array[..., i, j k])
```